Source/WebCore/PAL/pal/graphics/WebGPU/WebGPU.h

```diff
@@ -42,7 +42,7 @@ class GPU : public RefCounted<GPU> {
 public:
     virtual ~GPU() = default;
 
-    virtual void requestAdapter(const RequestAdapterOptions&, CompletionHandler<void(RefPtr<Adapter>&&)>&&) = 0;
+    virtual void requestAdapter(const RequestAdapterOptions&, CompletionHandler<void(RefPtr<Adapter>&&)>&&, unsigned = 0) = 0;
```

Comment:
Isn't there a better way to do this? Every `GPU` in the GPU process will come from a different web process, can't this information be in the `GPU` object (and not passed to requestAdapter())?

Reply:
We could store it in the GPU object instead. This would require from what I can tell storing the GPU object in the Adapter or the Device so we can get the handle.

Do you think storing a GPU& in the adapter is a good idea? I can do that if so

Reply:
I think that's the design we have already - child objects hold strong references to their parent object